service/share/discovery/discoverer.go

```diff
@@ -0,0 +1,75 @@
+package discovery
+
+import (
+	"context"
+	"errors"
+	"time"
+
+	logging "github.com/ipfs/go-log/v2"
+
+	"github.com/libp2p/go-libp2p-core/discovery"
+	"github.com/libp2p/go-libp2p-core/host"
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+const (
+	// PeersLimit is max amount of peers that will be discovered
+	PeersLimit = 5
+	// peerWeight total weight of discovered peers
+	peerWeight = 1000
+	// connectionTimeout is timeout given to connect to discovered peer
+	connectionTimeout = time.Minute
+)
+
+var log = logging.Logger("discovery")
+
+// Discoverer finds and stores discovered full nodes.
+type Discoverer struct {
+	set  *LimitedSet
+	host host.Host
+
+	discoverer discovery.Discoverer
+}
+
+func (d *Discoverer) Start(ctx context.Context) {
```

Comment:
we don't need this imo since it's not conforming to `Start` hook signature (i.e. `Start(ctx) error`)

just call `FindPeers` instead.

Reply:
Moved FindPeers to share.Service as we discussed.